src/controllers/authController.js

```diff
@@ -120,8 +122,47 @@ const logout = async (req, res) => {
   */
 };
 
+//forgotPassword endpoint
+const forgotPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw new CustomError.BadRequestError('Please provide an email');
+  }
+
+  const user = await User.findOne({ email });
+
+  if (!user) {
+    throw new CustomError.NotFoundError('User not found');
+  }
+
+  // Generate a random token
+  const resetToken = crypto.randomBytes(20).toString('hex');
+  const resetExpires = Date.now() + 15 * 60 * 1000; // 15 minutes from now
+
+  // Update user's reset token and expiration date
+  user.passwordResetToken = resetToken;
+  user.passwordResetExpiresOn = new Date(resetExpires);
+  await user.save();
+
+  // Send reset password email
+  const emailTemplate = 'forgotPassword'; // to adjust this based on our email template name
+  const recipient = user.email;
+  const localVariables = {
+    resetToken,
+  };
+  await emailSender.sendForgotPasswordEmail(
+    emailTemplate,
+    recipient,
+    localVariables
```

Comment:
You're passing in an object `{ resetToken: ... }` but your code in sender.js is expecting just a single string value for the resetToken as the second argument.

Reply:
Edited commit b0d0cd7fb1672665e7d2ae532d987ca63a10c327
Could you please check if I have done correctly? Thank you